### PR TITLE
add containerPilot for healthcheck

### PR DESCRIPTION
**Pull request by freignat91**
added containerPilot to register to Consul.
Need env. variable CONSUL=consulAddr:consulPort, if not exists then container pilot is inactive

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,9 +1,10 @@
 FROM alpine:3.3
 MAINTAINER Nicolas Degory <ndegory@axway.com>
 
+#Need to keep curl and bash after installation for containerPilot
 RUN apk update && \
-    apk --no-cache add python ca-certificates && \
-    apk --virtual envtpl-deps add --update py-pip python-dev curl && \
+    apk --no-cache add python ca-certificates curl bash && \
```
Comment (ndegory) [inline Dockerfile]: LGTM  curl was permanently added later in the Dockerfile, but we can keep it here with bash if it's a prerequisite for all images. I just want to make sure we have the same layer defined here between all TICK images
but then we have to remove the apk add curl later in the Dockerfile

File: Dockerfile [hunk 2]
```diff
@@ -39,6 +40,18 @@ ENV PRE_CREATE_DB **None**
 ENV SSL_SUPPORT **False**
 ENV SSL_CERT **None**
 
+
+# Add ContainerPilot
+RUN curl -Lo /tmp/cb.tar.gz https://github.com/joyent/containerpilot/releases/download/2.1.0/containerpilot-2.1.0.tar.gz \
```
Comment (ndegory) [inline Dockerfile]: can we use a ENV CONTAINERPILOT 2.1.0 before this block?

File: Dockerfile [hunk 3]
```diff
@@ -53,7 +66,7 @@ EXPOSE 8086
 
 VOLUME ["/data"]
 
-CMD ["/run.sh"]
+CMD ["sh", "-c", "/start.sh"]
```
Comment (ndegory) [inline Dockerfile]: actually, "sh -c" is already the default entry point, the command should only be "/start.sh", else we would have sh -c sh -c start.sh

File: Dockerfile [hunk 4]
```diff
@@ -39,6 +40,18 @@ ENV PRE_CREATE_DB **None**
...
+
+# Add ContainerPilot
+RUN curl -Lo /tmp/cb.tar.gz https://github.com/joyent/containerpilot/releases/download/2.1.0/containerpilot-2.1.0.tar.gz \
+&& tar -xz -f /tmp/cb.tar.gz \
+&& mv ./containerpilot /bin/
+COPY containerpilot.json /etc/containerpilot.json
+COPY start.sh /start.sh
+RUN chmod +x /start.sh
+
+ENV CONSUL=consul:8500
```
Comment (ndegory) [inline Dockerfile]: can we remove the CONSUL env variable by default? Else I'll have an issue using it in other contexts

File: start.sh [hunk 5]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+
+#display env. variables
+echo ---------------------------------------------------------------------------
+echo "CONSUL: "$CONSUL
+if [ -z "$CONSUL" ]; then
+  exec /run.sh
```
Comment (ndegory) [inline start.sh:7]: LGTM, that's what I'd like to see also on the telegraf image

File: start.sh [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+else
+  #update containerpilot conffile
+  sed -i "s/\[consul\]/$CONSUL/g" /etc/containerpilot.json
+  echo ---------------------------------------------------------------------------
+  echo containerPilot conffile
+  cat /etc/containerpilot.json
+  echo ---------------------------------------------------------------------------
+  /bin/containerpilot /run.sh
```
Comment (ndegory) [inline start.sh]: can we add an exec in front, to make sure it's the init process?

File: Dockerfile (hunk 1 above)
Comment (freignat91) [inline Dockerfile]: Curl is going to be used to check the availability of InfluxDB on a better way than the current "pidof influxd". We can remove Curl and re-add it later when needed.

File: Dockerfile (hunk 2 above)
Comment (freignat91) [inline Dockerfile]: yes sure

File: Dockerfile (hunk 1 above)
Comment (ndegory) [inline Dockerfile]: @freignat91 let's keep curl by default, we need it also for customization of most of the images in the init script

File: start.sh (hunk 6 above)
Comment (freignat91) [inline start.sh]: yes, unlike on telegraf it ok here.

**Comment by ndegory** [comment]
LGTM

**Pull request merged**